utils/src/check/super.c

```diff
@@ -65,15 +68,53 @@ static int check_super_flag(__le64 flags)
 	return 0;
 }
 
-static int check_quorum_config(struct scoutfs_quorum_config qconf)
+/*
+ * After checking the Quorum config we check the Quorum slots of the meta device
+ * for addressing issues
+ */
+static int check_quorum_slots(struct scoutfs_quorum_slot *slots)
 {
-	uint64_t qversion = le64_to_cpu(qconf.version);
+	uint16_t family;
+	int port;
+	struct in_addr in;
+
+	for (int i = 0; i < SCOUTFS_QUORUM_MAX_SLOTS; i++) {
+		port = le16_to_cpu(slots[i].addr.v4.port);
+		debug("Quorum slot %u port: %d\n", i, port);
+		if ((port <= 0) || (port > USHRT_MAX))
+			problem(PB_QSLOT_BAD_PORT, "Quorum Slot %u has bad port", i);
+
+		family = le16_to_cpu(slots[i].addr.v4.family);
+		debug("Quorum slot %u family: %u\n", i, family);
+		if (family != SCOUTFS_AF_IPV4)
+			problem(PB_QSLOT_BAD_FAM, "Port %u has unsupported family %u", port, family);
+
+		in.s_addr = htonl(le32_to_cpu(slots[i].addr.v4.addr));
+		debug("Quorum Slot %u ipv4 address: %s\n", i, inet_ntoa(in));
+		if ((in.s_addr & 0xff) != 0xff)
+			problem(PB_QSLOT_BAD_ADDR, "Quorum Slot %u has multicast address", i);
+		else if ((in.s_addr & 0xff000000) != 0)
+			problem(PB_QSLOT_NO_ADDR, "Quorum Slot %u has reserved address, cannot start with 0", i);
```

Comment:
s/NO/BAD/ here?

Reply:
??

Reply:
If the leftmost byte is zero, the IP address looks like `0.x.x.x`. That's a bad address since it's reserved.

Because you haven't checked (here) yet that x.x.x is or is not 0.0.0, this is a little confusing. In essence, you should check for 0.0.0.0 first. Then after that, check for 0.x.x.x and check for x.x.x.255. So changing the order here is also needed.